### PR TITLE
Adapted quiz statistics bar labels to the behaviour of exam scores

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [X] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

Requested by @krusche 

### Description

The code for placing the labels on the bar charts of the various quiz statistics was different from the one used in the exam scores modules. It was decided to adopt the exam scores code for the quiz statistics to create a uniform codebase.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create Quizzes, Participate in Quizzes
4. Check the various statistics available for quizzes for correct display(reachable under statistics in the exercise submenu)
image

### Screenshots

image
image

### CODE REVIEW

File: src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts [hunk 1]
```diff
@@ -79,52 +80,21 @@ export function createAnimation(dataSetProvider: DataSetProvider): ChartAnimatio
...
 
+            ctx.font = Chart.helpers.fontString(Chart.defaults.global.defaultFontSize, Chart.defaults.global.defaultFontStyle, Chart.defaults.global.defaultFontFamily);
+            ctx.textAlign = 'center';
+            ctx.textBaseline = 'bottom';
             const participants = dataSetProvider.getParticipants();
-            dataSetProvider.getDataSets().forEach((dataset, i) => {
-                const meta = chart.getDatasetMeta(i);
-                meta.data.forEach((bar: any, index) => {
+
+            dataSetProvider.getDataSets().forEach(function (dataset: DataSet, i: number) {
```
Comment (TobiasPr) [inline src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts:91]:
```suggestion
            dataSetProvider.getDataSets().forEach((dataset: DataSet, i: number) =>  {
```

Comment (TobiasPr) [reply src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts:91]: maybe also rename `i` to `index`

File: src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts [hunk 2]
```diff
@@ -79,52 +80,21 @@ export function createAnimation(dataSetProvider: DataSetProvider): ChartAnimatio
...
+                const meta = chartInstance.getDatasetMeta(i);
+                meta.data.forEach(function (bar: any, index: number) {
```
Comment (TobiasPr) [inline src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts:93]:
```suggestion
                meta.data.forEach((bar: any, index: number) => {
```

File: src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts [hunk 3]
```diff
@@ -136,6 +106,15 @@ export interface DataSetProvider {
     getParticipants(): number;
 }
 
+export function calculateTickMax(datasetProvider: DataSetProvider) {
+    const data = datasetProvider.getDataSets().map((dataset) => {
+        return dataset.data;
+    });
+    const merged = [].concat.apply([], data);
+    const max = Math.max(...merged);
```
Comment (TobiasPr) [inline src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts:114]: what are you doing here, can't we jsut do `Math.max(...data)` (do we need the `[].concat.apply([],data)`)

Comment (fde312) [reply src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts:114]: should then be `Math.max([...data])` I think.

**Review by fde312 — APPROVED** [review]
I tested it and works fine, good job! 🚀 
image

Code looks good too, just take a look on the comments from @TobiasPr.

File: src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts (hunk 3 above)
Comment (stefanwaldhauser) [reply src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts:114]: Hi the type of `data` is 
image so we need to flatten it first one leve

Comment (stefanwaldhauser) [reply src/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts]: I modified the code a little to make it more clearer

**Review by fde312 — CHANGES_REQUESTED** [review]
@stefanwaldhauser You are having a build error:


ERROR in [96msrc/main/webapp/app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component.ts[0m:[93m113[0m:[93m33[0m - [91merror[0m[90m TS2769: [0mNo overload matches this call.
build	01-Sep-2020 14:10:30	  Overload 1 of 2, '(...items: ConcatArray<never>[]): never[]', gave the following error.
build	01-Sep-2020 14:10:30	    Argument of type 'number[]' is not assignable to parameter of type 'ConcatArray<never>'.
build	01-Sep-2020 14:10:30	      The types returned by 'slice(...)' are incompatible between these types.
build	01-Sep-2020 14:10:30	        Type 'number[]' is not assignable to type 'never[]'.
build	01-Sep-2020 14:10:30	          Type 'number' is not assignable to type 'never'.
build	01-Sep-2020 14:10:30	  Overload 2 of 2, '(...items: ConcatArray<never>[]): never[]', gave the following error.
build	01-Sep-2020 14:10:30	    Argument of type 'number[]' is not assignable to parameter of type 'ConcatArray<never>'.

**Review by fde312 — APPROVED** [review]
Works fine for me and code looks good as well! Good job 🚀

**Review by fde312 — APPROVED** [review]

**Review by jpbernius — APPROVED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**